libc/src/__support/CPP/cstddef.h

```diff
@@ -16,6 +16,10 @@ namespace LIBC_NAMESPACE::cpp {
 
 enum class byte : unsigned char {};
 
+// When built with clang, this will use the definition from the clang header
+// __stddef_max_align_t.h
+using ::max_align_t LIBC_USING_IF_EXISTS;
```

Comment:
is this going to work? I don't think the compiler's `stddef.h` is included in this file so I'm not sure where this would be coming from.

Reply:
Can we just `#include <stddef.h> // for max_align_t` whenever we need it?  Is there any environment that does not provide `max_align_t` in `stddef.h`?